### PR TITLE
Adds OS and version to output if unsupported OS is detected

### PR DESCRIPTION
**Pull request by yubiuser**
Signed-off-by: Christian König <ckoenig@posteo.de>

**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/contributing/).

---
**What does this PR aim to accomplish?:**
Before:   [✗] Unsupported OS detected
After:  [✗] Unsupported OS detected: Linux Mint 19.3

**How does this PR accomplish the above?:**
Re-uses `$detected_os_pretty`

**What documentation changes (if any) are needed to support this PR?:**
No.

---
* You must follow the template instructions. Failure to do so will result in your pull request being closed.
* Please respect that Pi-hole is developed by volunteers, who can only reply in their spare time.

### CODE REVIEW

**Comment by cgzones** [comment]
Many thanks :+1:

**Comment by PromoFaux** [comment]
Oh, sorry @yubiuser, I didn't see this one, else i wouldn't have added the extra stuff to #3582 (easily changed)

This is a lot less convoluted code-wise than what I am doing (which is basically just the same as the OS detection output in the debug script)

**Comment by PromoFaux** [comment]
Would you mind re-doing this against the 5.1.2 branch instead of development, please? It'll need a rebase because there are some commits in dev not ready to go to release yet. 

Either that or just start again!

**Comment by yubiuser** [comment]
@PromoFaux 
I can do a re-base/start fresh on 5.1.2, no problem.
I like the idea of skipping the whole OS detection if `PIHOLE_SKIP_OS_CHECK=true`, which I haven't included in my PR. 
If you devs agree on that, I'll include it as well.

**Comment by PromoFaux** [comment]
Yeah, happy for you to reproduce that part in your PR, just make that the first check the function does.

I think we're still deciding on the wording for the examples

**Comment by dschaper** [comment]
Can we keep the PRs separate? I'd like to have one change per PR. If something needs to revert then it's easier to pull out smaller parts and leave the things that are working.

**Comment by PromoFaux** [comment]
Fair comment. I'll do my bit and close #3582

File: gravity.sh [hunk 1]
```diff
@@ -359,9 +359,10 @@ gravity_DownloadBlocklists() {
   for ((i = 0; i < "${#sources[@]}"; i++)); do
     url="${sources[$i]}"
     domain="${sourceDomains[$i]}"
+    id="${sourceIDs[$i]}"
 
     # Save the file as list.#.domain
-    saveLocation="${piholeDir}/list.${i}.${domain}.${domainsExtension}"
+    saveLocation="${piholeDir}/list.${id}.${domain}.${domainsExtension}"
```
Comment (dschaper) [inline gravity.sh:365]: I think the branch changes may have confused `git` a little bit too much.

Comment (yubiuser) [reply gravity.sh:365]: Yeah, I noticed. 
I think I need a proper git introduction... first time rebasing. Will close this one and make a new PR based on release/v5.1.2

**Comment by yubiuser** [comment]
PR based on release/v5.1.2

File: gravity.sh (hunk 1 above)
Comment (dschaper) [reply gravity.sh:365]: No worries, it's `git` so expect the unexpected.

**Pull request closed without merging**